frontend/src/app/item/listagem-page-item/listagem-page-item.component.ts

```diff
@@ -0,0 +1,146 @@
+import { ItemService } from './../../services/item.service';
+import { Component, OnInit, ViewChild, ElementRef } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { PageNotificationService } from '@nuvem/primeng-components';
+import { finalize } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-listagem-page-item',
+  templateUrl: './listagem-page-item.component.html',
+  styleUrls: ['./listagem-page-item.component.css']
+})
+export class ListagemPageItemComponent implements OnInit {
+
+    @ViewChild('inputFile', {static:false}) inputFile: ElementRef;
+    itens: any[] = [];
+    categorias: any[] = [];
+    form: FormGroup;
+    submit: boolean = false;
+    isEditing: boolean = false;
+    imageUrl: any;
```

Comment:
se é uma image URL é uma string, se não for acertar o nome da variavel e usar o blob.

Reply:
então TIto, essa imagemUrl foi a unica que não mudei, essa variavel recebe um FileReader.result que é do tipo (string | ArrayBuffer), e quando seto a variavel desse tipo ela não me deixar usar a função split que é a função que quebro ela para poder salvar no banco.